docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md

```diff
@@ -0,0 +1,416 @@
+
+In summary, this design eliminates the issues associated with existing Two-Level Scheduler Architecture, Siloed Multi-Scheduler Architecture, and Shared-State Multi-Scheduler Architecture. It renders good scalability, low latency, and global optimal scheduling while satisfying the user-supplied placement constraints.
+
+On the southbound, it integrates with different types of clusters via a flexible plugin design.
+
+Figure.1 is an illustration of the high-level architecture.
+![image.png](/images/2.3.png)
+
+In the future, we may explore whether this two-level architecture may flatten into a one-level architecture.
+
+## 2.4 Architecture Component Description
+
+The following table describes each component/subsystem in the Global Scheduling Architecture.
+
+<table>
+  <tr>
+    <th>Subsystem or Component</th>
+    <th>Description</th>
+    <th>Roles and Responsibilities</th>
+  </tr>
+  
+  <tr>
+    <td>API Server</td>
+    <td>
+    	API server exposes VM/Container CRUD APIs to clients, who use the Global Scheduler Service to request/read/update/delete VM/Container resource
+    </td>
+    <td>
+      <li>Receive, authenticate, authorize and validate API request and perform various admission controls</li>
+      <li>Perform data transformation and persist VM/Container information in Database cluster</li>
+      <li>Act as a proxy between the Request Distributor and the Database cluster, e.g., watch for VM request addition to DB and inform Request Distributor of this new VM request</li>
+      <li>Support multiple API Server partitions for scalability</li>
+      <li>The API Server design will reuse existing K8S declaration-based API server design and leverage Arktos API server scalability and multi-tenancy design/implementation</li>
+    </td>
+  </tr>
+  
+  <tr>
+    <td>Data Store for Static VM/Container Info, Scheduling Policy Info, Cluster's Geolocation/Region/AZ</td>
+    <td>Distributed database to store the VM/Container request profile and states, scheduling policy, each cluster's geolocation, region, AZ, supported VM flavor list. ETCD will be used for its consistency, persistency, and high availability. ETCD serves as the distributed, high availabe, multi-version data store with strong consistency, as well as messaging and/or eventing machanism for the system, i.e., when a key value changed, it delivers the changes to components which “watch” the change events for the key.</td>
+    <td>
+      <li>Distribute data store for high availability</li>
+      <li>Support Data Persistency</li>
+      <li>Support both strong and eventual data consistency</li>
+      <li>Support multiple partitions for scalability</li>
+      <li>Notify object changes to API server which will notify the VM/Container Request Distributor component</li>
+    </td>
+  </tr>
+  
+  <tr>
+    <td>Dynamic Cluster Resource Info Cache</td>
```

Comment:
Dynamic Cluster Resource Info Cache:We have considered to use redis as a  memory cache before, but if we chose etcd now , may be we need to consider the performance which is often be challenged with etcd.

Reply:
Database selection can be redis or any other cache DB. The interface API to Redis and ETCD are different, but the general design is the same.

Reply:
The goals of  global scheduler is manager 1K cluster include the  edge and region , support 100 QPS request. So ,we can make this estimate:
1、100 QPS resquest need  100/s full table query or condition query.
2、1K+cluster  update info per min,so need 1K write/s.